builds.go

```diff
@@ -0,0 +1,431 @@
+type BuildScope int
+
+const (
+	// BuildPending is one of five builds status.
+	BuildPending BuildScope = 1 << iota
+	// BuildRunning is one of five builds status.
+	BuildRunning
+	// BuildFailed is one of five builds status.
+	BuildFailed
+	// BuildSuccess is one of five builds status.
+	BuildSuccess
+	// BuildCanceled is one of five builds status.
+	BuildCanceled
+)
+
+func buildScope(scope BuildScope) string {
+	switch scope {
+	case BuildPending:
+		return "pending"
+	case BuildRunning:
+		return "running"
+	case BuildFailed:
+		return "failed"
+	case BuildSuccess:
+		return "success"
+	case BuildCanceled:
+		return "canceled"
+	default:
+		return "null"
+	}
+}
+
+// ListBuildsOptions are options for two list apis
+type ListBuildsOptions struct {
+	Scope *[]string `url:"scope, omitempty", json:"scope, omitempty"`
+}
+
+// BuildListBuildsOptions builds ListBuildsOptions from scopes in the form of integer.
+//
+// For example, BuildListBuildsOptions(BuildPending | BuildRunning | BuildFailed) will build
+// an option with "pending", "running" and "failed" scopes.
+func BuildListBuildsOptions(scopes int) *ListBuildsOptions {
+	opts := &ListBuildsOptions{
+		Scope: &[]string{},
+	}
+
+	if scopes == 0 {
+		return nil
+	}
+
+	for i := BuildPending; i <= BuildCanceled; i = i << 1 {
+		if (scopes & int(i)) != 0 {
+			temp := append(*opts.Scope, buildScope(i))
+			opts.Scope = &temp
+		}
+	}
+
+	if len(*opts.Scope) == 0 {
+		return nil
+	}
+
+	return opts
+}
+
+// BuildsService handles communication with the ci builds related methods
+// of the GitLab API.
+//
+// GitLab API docs: http://docs.gitlab.com/ce/api/builds.html
+type BuildsService struct {
+	client *Client
+}
+
+// Build represents a ci build.
+//
+// GitLab API docs: http://docs.gitlab.com/ce/api/builds.html
+type Build struct {
+	Commit *struct {
+		AuthorEmail string     `json:"author_email"`
+		AuthorName  string     `json:"author_name"`
+		CreatedAt   *time.Time `json:"created_at"`
+		ID          string     `json:"id"`
+		Message     string     `json:"message"`
+		ShortID     string     `json:"short_id"`
+		Title       string     `json:"title"`
+	} `json:"commit"`
+	Coverage      *string    `json:"coverage, omitempty"`
+	CreatedAt     *time.Time `json:"created_at"`
+	ArtifactsFile *struct {
+		Filename string `json:"filename"`
+		Size     int    `json:"size"`
+	} `json:"artifacts_file"`
+	FinishedAt *time.Time `json:"finished_at"`
+	ID         int        `json:"id"`
+	Name       string     `json:"name"`
+	Ref        string     `json:"ref"`
+	Runner     *struct {
+		ID          int    `json:"id"`
+		Description string `json:"description"`
+		Active      bool   `json:"active"`
+		IsShared    bool   `json:"is_shared"`
+		Name        string `json:"Name"`
+	} `json:"runner, omitempty"`
+	Stage     string     `json:"stage"`
+	StartedAt *time.Time `json:"started_at"`
+	Status    string     `json:"status"`
+	Tag       bool       `json:"tag"`
+	User      *struct {
```

Comment:
I think we should use the existing `User` struct here: https://github.com/xanzy/go-gitlab/blob/master/users.go#L36